cellmlmanip/units.py

```diff
@@ -290,6 +290,10 @@ def convert(self, quantity, unit):
         """
         assert isinstance(quantity, self._registry.Quantity)
         assert isinstance(unit, self._registry.Unit)
+        #  Trying to convert dimensionless gives an error but we can convert to it
+        if quantity.units == self._registry.dimensionless:
+            quantity, unit = 1 * unit, quantity.units
+            return 1 / quantity.to(unit)
```

Comment:
Indeed it looks like a pint error, you can convert one way, but not the other way.
actually I think it should be `quantity.magnitude / quantity.to(unit)` rather than `1 / quantity.to(unit)`
so that 2 meters become 4 half meters en 3 half meters become 1.5 meter i.e.
```
    def test_dimensionless_multiplier2(self):
        model = load_model('dimensionless_multiplier2.cellml')
        meter = model.units.get_unit('meter')
        half_meter = model.units.get_unit('half_meter')
        cf1 = model.units.convert(2 * meter, half_meter).magnitude
        cf2 = model.units.convert(3 * half_meter, meter).magnitude
        assert cf1 == 4.0
        assert cf2 == 1.5
```